lib/module-index-client/src/types.rs

```diff
@@ -75,3 +73,40 @@ pub struct ExtraMetadata {
     pub schemas: Vec<String>,
     pub funcs: Vec<FuncMetadata>,
 }
+
+#[derive(Deserialize, Serialize, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct ListLatestModulesRequest {
+    pub hashes: Vec<String>,
+}
+
+#[derive(Deserialize, Serialize, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct ListLatestModulesResponse {
+    pub modules: Vec<LatestModuleResponse>,
+}
+
+/// This struct is nearly the same as the [`ModuleDetailsResponse`], but it does not include `past_hashes` since the
+/// data is unneeded and requires additional query logic.
+#[derive(Deserialize, Serialize, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct LatestModuleResponse {
+    pub id: String,
+    pub name: String,
+    pub description: Option<String>,
+    pub owner_user_id: String,
+    pub owner_display_name: Option<String>,
+    pub metadata: serde_json::Value,
+    pub latest_hash: String,
+    pub latest_hash_created_at: DateTime<Utc>,
+    pub created_at: DateTime<Utc>,
+    pub schema_id: Option<String>,
+}
```

Comment:
We probably do not need all this data, but the query can easily access it. We can likely reduce the amount of data sent over the wire once we have the update mechanism in place.